### PR TITLE
Fix ETH gas fees

### PR DESCRIPTION
**Pull request by tomasklim**
## Description

- fees bubble was jumping when fee was calculated
- in case of sending ERC-20 tokens, the value of gas limit was changing constantly between correct amount and 21000 while changing amount input
- gas limit was not calculated correctly when using send max as amount was set to 1 wei because amount is not yet calculated and is based on the result of the function `getEthereumEstimateFeeParams`
Screenshot 2023-06-07 at 16 34 19
which can result into failing txs if the contract is conditional based on amount

## Related Issue

I fixed it during ETH staking. I can put it there, but probably better to separate it to make it clearer

## Screenshots:

https://github.com/trezor/trezor-suite/assets/33235762/94a922a6-28f2-45f5-ae12-070c0a8eb44b

### CODE REVIEW

**Comment by sime** [comment]
@tomasklim for ETH sending the gas limit will always be the same?

Not sure what you mean by second place.

**Comment by Hannsek** [comment]
We can rephrase it a bit in eth flow. 

_Fee_ -> _Maximum fee_

In FW, it will be like this: 
image

**Comment by tomasklim** [comment]
> @tomasklim for ETH sending the gas limit will always be the same?
> 
> Not sure what you mean by second place.

@sime 
1) According to docs yes, according to second picture in which I want to send ETH to contract, it doesn't look like that. The limit increased to 21062 after filling.

2) I swapped gas limit with gas price. The order
From:
Screenshot 2023-06-07 at 12 16 26

To:
Screenshot 2023-06-07 at 12 16 42

**Comment by Hannsek** [comment]
I would have gas limit on the first place.

**Comment by tomasklim** [comment]
Okay, I will put there some min-width instead to fix this jumping 

https://github.com/trezor/trezor-suite/assets/33235762/9cfc6e2b-3177-4d04-aa81-5a9c9a005305

**Comment by Hannsek** [comment]
Thanks, however, isn' the gas limit behaviour weird? I thought that we use the "old"process and the gas limit is always 21000 units.

**Comment by tomasklim** [comment]
It depends where you send and what you send. It has never been 21000 for tokens.

EDIT: 
order changed back

**Comment by Hannsek** [comment]
Aha, forgot we talked about tokens here.

File: packages/suite/src/actions/wallet/send/sendFormEthereumActions.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ export const composeTransaction =
                     ...getEthereumEstimateFeeParams(
                         address || account.descriptor,
                         tokenInfo,
-                        amount,
+                        tokenInfo ? amount : amount || account.formattedBalance,
```
Comment (matejkriz) [inline packages/suite/src/actions/wallet/send/sendFormEthereumActions.ts]:
```suggestion
                        amount || (tokenInfo ? tokenInfo.balance : account.formattedBalance) // TODO comment
```

Comment (matejkriz) [reply packages/suite/src/actions/wallet/send/sendFormEthereumActions.ts]: and update getEthereumEstimateFeeParams (remove handling falsy amount from there)

File: packages/suite/src/components/wallet/Fees/components/FeeDetails.tsx [hunk 2]
```diff
@@ -69,24 +75,36 @@ const BitcoinDetails = ({ networkType, feeInfo, selectedLevel, transactionInfo }
...
-            </Label>
-            {`${selectedLevel.feePerUnit} ${getFeeUnits(networkType)}`}
-        </Item>
-    </Wrapper>
-);
+const EthereumDetails = ({ networkType, selectedLevel, transactionInfo }: Props) => {
+    const [fee, setFee] = useState<string | undefined>(selectedLevel.feeLimit);
+
+    useEffect(() => {
+        if (transactionInfo && transactionInfo.type !== 'error') {
```
Comment (matejkriz) [inline packages/suite/src/components/wallet/Fees/components/FeeDetails.tsx]: nit: would be nice to put it in constant and reuse on L#94

File: packages/suite/src/actions/wallet/send/sendFormEthereumActions.ts (hunk 1 above)
Comment (tomasklim) [reply packages/suite/src/actions/wallet/send/sendFormEthereumActions.ts]: ab5f6b2e6034dc9f61d7b5dc6cea77b1d7814a17

File: packages/suite/src/components/wallet/Fees/components/FeeDetails.tsx (hunk 2 above)
Comment (tomasklim) [reply packages/suite/src/components/wallet/Fees/components/FeeDetails.tsx]: fixed

**Review by matejkriz — APPROVED** [review]
👍

**Comment by bosomt** [comment]
**QA OK**

**Info:**
 - Suite version: web 23.7.0 (bd2f5aebe78af58c2bee5d8d49dd7d8bdbeb38f5)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/114.0
 - OS: MacIntel
 - Screen: 1824x1140
 - Device: model T 2.6.1 Universal (revision 089ff768229f543c89927ff34c4d45d830a2442f)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**